### PR TITLE
Add mdbook github action

### PR DESCRIPTION
**Pull request by tnachen**
Add github action that automatically pushes doc changes into github page on both pushes to next and master branch

Example running
https://github.com/tnachen/miden/runs/5749281007?check_suite_focus=true

### CODE REVIEW

File: .github/workflows/book.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: halo2 book
```
Comment (bobbinth) [inline .github/workflows/book.yml:1]: Probably need to update this :)

File: .github/workflows/book.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+on:
+  push:
+    branches:
+      - master
+      - next
+
+jobs:
+  deploy:
+    runs-on: ubuntu-18.04
```
Comment (bobbinth) [inline .github/workflows/book.yml:11]: Could we replace this with `ubuntu-latest`?

File: .github/workflows/book.yml [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          # Pinned until https://github.com/rust-lang/rust/issues/93476 is fixed.
+          toolchain: nightly-2022-01-18
```
Comment (bobbinth) [inline .github/workflows/book.yml:17]: Seems like this issue has been fixed already.

File: .github/workflows/book.yml [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+          override: true
+
+      - name: Setup mdBook
+        uses: peaceiris/actions-mdbook@v1
+        with:
+          mdbook-version: 'latest'
+
+      - name: Build halo2 book
```
Comment (bobbinth) [inline .github/workflows/book.yml:25]: Probably need to update this :)

**Review by bobbinth — CHANGES_REQUESTED** [review]
Thank you! Some comments inline.

File: .github/workflows/book.yml (hunk 4 above)
Comment (tnachen) [reply .github/workflows/book.yml:25]: fixed

File: .github/workflows/book.yml (hunk 2 above)
Comment (tnachen) [reply .github/workflows/book.yml:11]: fixed

File: .github/workflows/book.yml (hunk 1 above)
Comment (tnachen) [reply .github/workflows/book.yml:1]: fixed

**Comment by tnachen** [comment]
Tested here https://github.com/tnachen/miden/actions/runs/2066973095

File: .github/workflows/book.yml (hunk 3 above)
Comment (tnachen) [reply .github/workflows/book.yml]: fixed

File: .github/workflows/book.yml [hunk 5]
```diff
@@ -0,0 +1,32 @@
+name: miden book
+
+on:
+  push:
+    branches:
+      - master
```
Comment (bobbinth) [inline .github/workflows/book.yml:6]: missed this on the first review: we don't have a 'main` branch instead of `master`. Should it be fixed here?

**Review by bobbinth — CHANGES_REQUESTED** [review]
Thank you! One more comment inline.

**Review by bobbinth — APPROVED** [review]

**Pull request merged**